src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -544,24 +554,123 @@ void MarkCustomAttributes (ICustomAttributeProvider provider, in DependencyInfo
 				MarkCustomAttribute (ca, reason);
 				MarkSpecialCustomAttributeDependencies (ca, provider);
 			}
+
+			if (!(provider is MethodDefinition || provider is FieldDefinition))
+				return;
+
+			var dynamicDependencies = _context.Annotations.GetLinkerAttributes<DynamicDependency> ((IMemberDefinition) provider);
+			Debug.Assert (dynamicDependencies != null);
```

Comment:
What is this assert for ?

Reply:
Just to assert that the API doesn't give back null. :) I can remove it.